src/dune/simple_rules.ml

```diff
@@ -35,6 +35,16 @@ let check_filename =
       Path.as_in_build_dir_exn p
     | Dir p -> not_in_dir ~error_loc (Path.to_string p)
 
+let add_alias sctx ~dir ~name ~stamp ~loc ?(locks = []) build =
+  let alias = Alias.make name ~dir in
+  SC.add_alias_action sctx alias ~dir ~loc ~locks ~stamp build
+
+let stamp ~deps ~action ~extra_bindings =
+  ( "user-alias"
+  , Bindings.map ~f:Dep_conf.remove_locs deps
+  , Option.map ~f:Action_unexpanded.remove_locs action
+  , Option.map extra_bindings ~f:Pform.Map.to_stamp )
```

Comment:
While we're at this, do you know what `extra_bindings` are and why they're used here? I know it's a pre-existing thing, but it seems weird to me.

Reply:
One example where it's used is to introduce the `%{test}` binding for the `test` stanza.